code/obj/machinery/door/door_control.dm

```diff
@@ -602,6 +603,9 @@ ADMIN_INTERACT_PROCS(/obj/machinery/door_control, proc/toggle)
 						M.operating = 0
 			M.setdir()
 
+	if(src.single_use)
+		qdel(src)
```

Comment:
Should return after qdel - no need to handle cooldowns etc if it's getting deleted